source/GarminSDComms.mc

```diff
@@ -63,16 +63,15 @@ class GarminSDComms {
 
   function sendAccelData() {
     var tagStr = "SDComms.sendAccelData()";
-    var dataObj = mAccelHandler.getDataJson();
-    if (mDataRequestInProgress) {
+    var waitingTime = Time.now().subtract(mDataSendStartTime);
+    if ((mDataRequestInProgress) && (waitingTime.lessThan(TIMEOUT))){
```

Comment:
Ok, so we send data if there is not a request in progress, or if the request has timed out.  That sounds reasonable.
Have you tested it to confirm it works ok?

Reply:
Ok, thanks - if it is working on yours, will merge this into the develop branch and I'll give it a test.